src/platform/ESP32/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -0,0 +1,319 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/support/CodeUtils.h>
+#include <lib/support/SafeInt.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/ESP32/ESP32Utils.h>
+#include <platform/ESP32/NetworkCommissioningDriver.h>
+
+#include "esp_wifi.h"
+
+#include <limits>
+#include <string>
+
+using namespace ::chip;
+
+namespace chip {
+namespace DeviceLayer {
+namespace NetworkCommissioning {
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
```

Comment:
Could we include this C++ source file in the compilation if wifi is enabled? That would avoid the need to have ifdef-guards in the source and always compile.

Reply:
Have modified the gn file to include this file if wifi is enabled